### PR TITLE
Email notifications

### PR DESCRIPTION
**Pull request by kvikshaug**
I am currently using my personal account on Sendgrid, so we should probably create a shared team account.

Depends on https://github.com/DD-DeCaF/iam/pull/19

### CODE REVIEW

File: src/metabolic_ninja/tasks.py [hunk 1]
```diff
@@ -117,13 +122,31 @@ def predict(self, model_obj, product_name, max_predictions, aerobic,
     #  default_biomass_reaction. Maybe we need a new field for the medium
     #  database model?
     model.carbon_source = "EX_glc__D_e"
+    # Retrieve user details. This should be retrieved here, to ensure that the
```
Comment (Midnighter) [inline src/metabolic_ninja/tasks.py:124]: In terms of code organization, it seems to me like this new section should go into the `notify` task.

File: src/metabolic_ninja/tasks.py [hunk 2]
```diff
@@ -117,13 +122,31 @@ def predict(self, model_obj, product_name, max_predictions, aerobic,
...
+    organism_name = response.json()['name']
     # Define the workflow.
     workflow = (
         find_product.s(product_name, source) |
         find_pathways.s(model, max_predictions, source) |
         optimize.s(model) |
         concatenate.s() |
-        persist.s(job_id)
+        persist.s(job_id) |
+        notify.s(job_id, product_name, organism_id, user_name, user_email,
```
Comment (Midnighter) [inline src/metabolic_ninja/tasks.py]: If you use the immutable signature instead, celery will not pass on the previous result. You can thus avoid accepting `result` on line 408 below and also don't need to return it again.

```suggestion
        notify.si(job_id, product_name, organism_id, user_name, user_email,
```

File: src/metabolic_ninja/tasks.py [hunk 3]
```diff
@@ -379,3 +402,32 @@ def persist(result, job_id):
...
     return result
+
+
+@celery_app.task()
+def notify(result, job_id, product_name, organism_id, user_name, user_email,
+           organism_name):
+    try:
+        sendgrid = SendGridAPIClient()
+        mail = Mail()
+        mail.from_email = Email("DD-DeCaF <hello@dd-decaf.eu>")
```
Comment (Midnighter) [inline src/metabolic_ninja/tasks.py]: How about a slightly different e-mail address?
```suggestion
        mail.from_email = Email("DD-DeCaF <notifications@dd-decaf.eu>")
```

**Review by Midnighter — APPROVED** [review]
Some suggestions for changes. Take them as you see fit.

File: src/metabolic_ninja/tasks.py (hunk 1 above)
Comment (kvikshaug) [reply src/metabolic_ninja/tasks.py:124]: The reason it's not is that the forwarded JWT required to retrieve user details may have expired by the time the `notify` task executes

**Pull request merged**